public/js/src/api.js

```diff
@@ -0,0 +1,53 @@
+const POLITICS_URL = `https://content.guardianapis.com/search?from-date=${yesterday()}&to-date=${today()}&q=politics&api-key=test&show-fields=all`
+
+const frontPage = new FrontPage();
+
+fetchArticles();
+
+function fetchArticles(url = POLITICS_URL) {
+  if (ENVIRONMENT === 'test') {
```

Comment:
this was a good idea for getting around the fact that you couldn't use any libraries for mocking! Generally we want to avoid putting such test-specific behaviour into our code though. One other technique that might have worked is a technique "monkey patching". It would allow you to replace the `fetchArticles` function with a different function in your tests.

Reply:
Thanks Simo, this was a tricky area to know what to do. I will bear that approach in mind :)